explorer/src/lib/chain-info/transformTransaction.js

```diff
@@ -1,23 +1,45 @@
 import { unixTsToDate } from "$lib/dusk/date";
 
 /** @type {(v: GQLTransaction) => Transaction} */
-const transformTransaction = (tx) => ({
-  amount: undefined,
-  blockhash: tx.blockHash,
-  blockheight: tx.blockHeight,
-  date: unixTsToDate(tx.blockTimestamp),
-  feepaid: tx.gasSpent * tx.tx.gasPrice,
-  from: undefined,
-  gaslimit: tx.tx.gasLimit,
-  gasprice: tx.tx.gasPrice,
-  gasspent: tx.gasSpent,
-  memo: tx.tx.memo ?? "",
-  method: tx.tx.isDeploy ? "deploy" : (tx.tx.callData?.fnName ?? "transfer"),
-  success: tx.err === null,
-  to: undefined,
-  txerror: tx.err ?? "",
-  txid: tx.id,
-  txtype: tx.tx.txType,
-});
+const transformTransaction = (tx) => {
+  const payloadString = tx.tx.json ?? "{}";
+  const payload = JSON.parse(payloadString);
+
+  /** @type {Transaction} */
+  const transaction = {
+    blockhash: tx.blockHash,
+    blockheight: tx.blockHeight,
+    date: unixTsToDate(tx.blockTimestamp),
+    feepaid: tx.gasSpent * tx.tx.gasPrice,
+    gaslimit: tx.tx.gasLimit,
+    gasprice: tx.tx.gasPrice,
+    gasspent: tx.gasSpent,
+    memo: tx.tx.memo ?? "",
+    method: tx.tx.isDeploy ? "deploy" : (tx.tx.callData?.fnName ?? "transfer"),
+    payload,
+    success: tx.err === null,
+    txerror: tx.err ?? "",
+    txid: tx.id,
+    txtype: tx.tx.txType,
+  };
+
+  if (payload.value !== undefined) {
```

Comment:
Not that it should matter here, but this way you changed the type from having properties with a value that is possibly `undefined` to a type where the properties themselves are optional (meaning that the keys might not be there anymore).
This can be risky in general as you might break checks that relies on those keys elsewhere.

Reply:
Yes, yes, that I know: was just making sure you're aware of the difference. Shouldn't matter here, but still generically speaking I prefer to have always the same data shape rather than optional keys.